test/oas3/OperationTest.ts

```diff
@@ -190,6 +170,173 @@ describe('oas3 Operation', function() {
             expect(authenticated).to.eql({});
         });
 
+        describe('Security schemes combined via OR', async function() {
+            it('should return result of first successful authenticator', async function() {
+                const options = {
+                    authenticators: {
+                        basicAuth() {
+                            return BASIC_AUTH_SUCCESS_RESULT;
+                        },
+                        oauth() {
+                            return DEFAULT_OAUTH_RESULT;
+                        }
+                    }
+                };
+
+                const op: oas3.OperationObject = {
+                    responses: {200: {description: "ok"}},
+                    security: [
+                        {basicAuth: []},
+                        {oauth: ['admin']},
+                    ]
+                };
+
+                const operation: Operation = makeOperation('get', op, {options});
+                const authenticated = await operation.authenticate(this.exegesisContext);
+                expect(authenticated).to.exist;
+                expect(authenticated).to.eql({
+                    basicAuth: BASIC_AUTH_SUCCESS_RESULT
+                });
+            });
+
+            it('should authenticate an incoming request if one authenticator hard-fails', async function() {
+                const options = {
+                    authenticators: {
+                        basicAuth() {
+                            return {type: 'invalid'};
+                        },
+                        oauth() {
+                            return DEFAULT_OAUTH_RESULT;
+                        }
+                    }
+                };
+
+                const op: oas3.OperationObject = {
+                    responses: {200: {description: "ok"}},
+                    security: [
+                        {basicAuth: []},
+                        {oauth: ['admin']},
+                    ]
+                };
+
+                const operation: Operation = makeOperation('get', op, {options});
+                const authenticated = await operation.authenticate(this.exegesisContext);
+                expect(authenticated).to.exist;
```

Comment:
Yeah, I've been doing a lot of reading around the spec. 

To get the scenario that you want you would `AND` the authentication types. I included the info from the spec in the `docs/OAS3 Security.md` and how it works.

We should follow the`OR' convention and let it authentication succeed if at least one of the authenticators succeeds. 

By doing this users can decide which strategy they wish to follow.

Reply:
I kind of flip back and forth about how I feel about it.  :P  But it does seem like if you're providing, say, a username and a password, and the password is wrong, then even if you're providing some other credential that passes, we should probably reject, because something is amiss; maybe you're intending to do the action as the username/password and are unaware you're accidentally sending along some other credential...

I'm going to merge this for now, because I think it makes things better overall, and then I'm going to raise an issue against OpenAPI-Specification and ask for clarification.  :)

Reply:
Cheers Jason. 

Here is the link to the spec, for your reference, where it mentions `only one of Security Requirement Objects in the list needs to be satisfied to authorize the request`

```
When a list of Security Requirement Objects is defined on the Open API object or Operation Object, only one of Security Requirement Objects in the list needs to be satisfied to authorize the request.
```

https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#security-requirement-object

Reply:
Yeah, I've read it more than once, too.  :)  I've had this exact discussion with a co-worker, with a friend of mine who is a security researcher, and a few other times as well, and everyone has a different take on how this should work.  :)

Reply:
Yeah, I agree with the above.  I'd rather get the spec wrong and not give access when we should than go the other way around and give access when we shouldn't.  Thanks again for the help!